### PR TITLE
Add following validations based on azure storage account settings

### PR DESCRIPTION
**Pull request by bharathsm-ms**
1. Validate SMB versions
2. Validate autheniticate mechanism
3. Validate channel encryption settings
4. Notify user about type of encryption configured for Kerbros ticket
5. In case of SMB2.1, Check secure transfer option is enabled and notify user
6. Minor bug fixes and logs

### CODE REVIEW

**Review by sprasad-microsoft — CHANGES_REQUESTED** [review]

File: AzFileDiagnostics/Linux/AzFileDiagnostics.sh [hunk 1]
```diff
@@ -48,11 +53,225 @@ usage()
...
+	select opt in "${options[@]}"
+	do
+		case $opt in
+			"SMB2.1")
+				requested_ver="SMB2.1"
+				#Check azure portal configurations for secure transfer enable setting
+				az storage account show -n $ACCOUNT | grep "enableHttpsTrafficOnly" | grep -q "true"
+				retVal=$?
+				if [ $retVal -eq 0 ]; then
+					print_log "Secure transfer is enabled on server, Please disable to from azure portal to mount SMB2 shares" "error"
```
[the remark this answers is not in the document]
Comment (bharathsm-ms) [reply AzFileDiagnostics/Linux/AzFileDiagnostics.sh]: Done

File: AzFileDiagnostics/Linux/AzFileDiagnostics.sh [hunk 2]
```diff
@@ -48,11 +53,225 @@ usage()
...
+{
+	print_log "Validating Channel encryption settings"
+	supported_enc=$(echo $PROTOCOLSETTINGS | sed 's/,/\n/g' | grep "smbChannelEncryption" | awk -F\" '{print $(NF-1)}' | sed 's/;/ /g')
+
+	if [ $requested_ver = "SMB3.1.1" ]; then
+		default_enc="AES-128-GCM"
+		if [[ " $supported_enc " = *" $default_enc "* ]]; then
+			print_log "$default_enc Channel encryption supported by server for 3.1.1" "info"
+		elif  [[ " $supported_enc " = *" AES-256-GCM "* ]]; then
+			print_log "AES-256-GCM is supported by server for 3.1.1, but you should install cifs module on client with module parameter 'require_gcm_256' to use AES-256-GCM" "warning"
```
[the remark this answers is not in the document]
Comment (bharathsm-ms) [reply AzFileDiagnostics/Linux/AzFileDiagnostics.sh]: Changed accordingly and added a command to enable gcm_256

File: AzFileDiagnostics/Linux/AzFileDiagnostics.sh [hunk 3]
```diff
@@ -48,11 +53,225 @@ usage()
...
+		elif [[ " $supported_enc " = " AES-128-CCM " ]]; then
+			print_log "AES-128-CCM not supported with Azure SMB 3.1.1, Please enable other encryption mechanisms using https://docs.microsoft.com/en-us/azure/storage/files/files-smb-protocol?tabs=azure-portal#smb-security-settings" "error"
+			exit 1
+		else
+			print_log "Unknown Encryption mechanism" "error"
```
[the remark this answers is not in the document]
Comment (bharathsm-ms) [reply AzFileDiagnostics/Linux/AzFileDiagnostics.sh]: Done

File: AzFileDiagnostics/Linux/AzFileDiagnostics.sh [hunk 4]
```diff
@@ -48,11 +53,225 @@ usage()
...
+
+		#Check if user has already logged into azure cli
+		az account show  > /dev/null 2>&1
+		retVal=$?
+		if [ $retVal -ne 0 ]; then
+			print_log "Please follow the steps to login to azure cli, If you have trouble follow https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli and login manually" "info"
+			az login
+		else
+			print_log "Already logged into Azure command line interface, please verify details and proceed further" "info"
+			az account show
```
[the remark this answers is not in the document]
Comment (bharathsm-ms) [reply AzFileDiagnostics/Linux/AzFileDiagnostics.sh:283]: Done. The details gets printed when user does "az login". If we do "az account show" again then there will be duplicates.

File: AzFileDiagnostics/Linux/AzFileDiagnostics.sh [hunk 5]
```diff
@@ -627,7 +886,7 @@ disable_log()
 
 ## Prompt user to select if he wants to map drive or not.
 
-print_log "Script has validated the client settings and do you want to map drive by script?"
+print_log "Do you want to map drive by script?" "info"
```
[the remark this answers is not in the document]
Comment (bharathsm-ms) [reply AzFileDiagnostics/Linux/AzFileDiagnostics.sh]: Done

File: AzFileDiagnostics/Linux/AzFileDiagnostics.sh [hunk 6]
```diff
@@ -696,15 +955,18 @@ if [ "$DIAGON" -eq 0 ]; then
 fi
 
 
-command="mount -t cifs "$UNCPATH"  "$mountpoint" -o vers=3.0,username="$username",password=$password,dir_mode=0777,file_mode=0777,sec=ntlmssp"
-print_log "Try with mounting share using SMB3.0"
+command="mount -t cifs "$UNCPATH"  "$mountpoint" -o vers="$smb_ver",username="$username",password=$password,dir_mode=0777,file_mode=0777,sec="$sec_name""
```
[the remark this answers is not in the document]
Comment (bharathsm-ms) [reply AzFileDiagnostics/Linux/AzFileDiagnostics.sh]: Done

**Review by sprasad-microsoft — APPROVED** [review]

**Pull request merged**